snforge_std/src/byte_array.cairo

```diff
@@ -1,7 +1,21 @@
+use core::byte_array::BYTE_ARRAY_MAGIC;
+
 fn byte_array_as_felt_array(self: @ByteArray) -> Array<felt252> {
     let mut serialized = array![];
 
     self.serialize(ref serialized);
 
     serialized
 }
+
+fn try_deserialize_bytearray_error(x: Array<felt252>) -> Result<ByteArray, ()> {
+    if x.len() > 0 && *x.at(0) == BYTE_ARRAY_MAGIC {
+        let mut x_span = x.span().slice(1, x.len() - 1);
+        let deserialized = match Serde::<ByteArray>::deserialize(ref x_span) {
+            Option::Some(str) => str,
+            Option::None => panic!("panic string not deserializable")
```

Comment:
Panic in function returning `Result` is unexpected

Reply:
This should never happen, because it means that something is seriously wrong with the serialization/deserialization itself
Do you think it should still be handleable? I mean it can happen if a user calls it on a random array, but seems pretty good idea to crash then since it's usage error

Reply:
This function is literraly called `try_deserialize_...`. Why function trying to deserialize something and returning `Result` should panic when deserialization fails?